addons/google_account/google_account.py

```diff
@@ -116,21 +120,22 @@ def _refresh_google_token_json(self, cr, uid, refresh_token, service, context=No
             uri = self.get_uri_oauth(a='token')
 
             data = werkzeug.url_encode(params)
-            st, res, ask_time = self._do_request(cr, uid, uri, params=data, headers=headers, type='POST', preuri='', context=context)
+            st, res, ask_time = self._do_request(uri, params=data, headers=headers, type='POST', preuri='')
         except urllib2.HTTPError, e:
             if e.code == 400:  # invalid grant
                 registry = openerp.modules.registry.RegistryManager.get(request.session.db)
                 with registry.cursor() as cur:
-                    self.pool['res.users'].write(cur, uid, [uid], {'google_%s_rtoken' % service: False}, context=context)
+                    self.env.uid.write({'google_%s_rtoken' % service: False})
```

Comment:
Not going to work, `env.uid` is the id of the current user. You want `env.user` which is the actual `res.user` object.